File: bindings/lua/Polycode/JoystickInfo.lua
class "JoystickInfo"


function JoystickInfo:__getvar(name)
	if name == "deviceID" then
		return Polycode.JoystickInfo_get_deviceID(self.__ptr)
	elseif name == "deviceIndex" then
		return Polycode.JoystickInfo_get_deviceIndex(self.__ptr)
	end
end

function JoystickInfo:__setvar(name,value)
	if name == "deviceID" then
		Polycode.JoystickInfo_set_deviceID(self.__ptr, value)
		return true
	elseif name == "deviceIndex" then
		Polycode.JoystickInfo_set_deviceIndex(self.__ptr, value)
		return true
	end
	return false
end
function JoystickInfo:JoystickInfo(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.JoystickInfo(unpack(arg))
	end
end

function JoystickInfo:__delete()
	if self then Polycode.delete_JoystickInfo(self.__ptr) end
end

File: bindings/javascript/Polycode/JoystickInfo.js
function JoystickInfo() {
	Object.defineProperties(this, {
		'deviceID': { enumerable: true, configurable: true, get: JoystickInfo.prototype.__get_deviceID, set: JoystickInfo.prototype.__set_deviceID},
		'deviceIndex': { enumerable: true, configurable: true, get: JoystickInfo.prototype.__get_deviceIndex, set: JoystickInfo.prototype.__set_deviceIndex}
	})
}
JoystickInfo.prototype.__get_deviceID = function() {
	return Polycode.JoystickInfo__get_deviceID(this.__ptr)
}

JoystickInfo.prototype.__set_deviceID = function(val) {
	Polycode.JoystickInfo__set_deviceID(this.__ptr, val)
}

JoystickInfo.prototype.__get_deviceIndex = function() {
	return Polycode.JoystickInfo__get_deviceIndex(this.__ptr)
}

JoystickInfo.prototype.__set_deviceIndex = function(val) {
	Polycode.JoystickInfo__set_deviceIndex(this.__ptr, val)
}

